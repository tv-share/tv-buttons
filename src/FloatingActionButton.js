import React from "react";
import PropTypes from "prop-types";

import "./style/floatingActionButton";
/**
* Renders the SelectInput Component
* @param  {String}    	className is a string that will be attached to the select input's className (optional)
* @param  {String}      icon defines the path of the icon to be displayed inside the button
* @param  {Function}    onClick defines a function to be called whenever the button is pressed
* @param  {String}      fixed defines if button should be fixed in the bottom right of the screen
* @param  {Object}      events is an object with one or more React Synthetic Events
* @return {JSX}         SelectInput Component
*/
const FloatingActionButton = ({ className, icon, onClick, fixed = false, events = {} }) => {
	return (
		<div className={`tv-floating-action-button ${fixed ? "-fixed" : ""} ${className ? className : ""}`} icon={icon} onClick={onClick}>
			<img src={icon}/>
		</div>
	);
};

FloatingActionButton.propTypes = {
	icon: PropTypes.string.isRequired,
	onClick: PropTypes.func.isRequired,
	className: PropTypes.string,
	fixed: PropTypes.bool,
	events: PropTypes.object
};

export default FloatingActionButton;