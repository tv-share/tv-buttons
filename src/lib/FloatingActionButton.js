import React from "react";
import PropTypes from "prop-types";


/**
* Renders the SelectInput Component
* @param  {String}    	className is a string that will be attached to the select input's className (optional)
* @param  {String}      icon defines the path of the icon to be displayed inside the button
* @param  {Function}    onClick defines a function to be called whenever the button is pressed
* @param  {String}      position defines the relative position of the button
* @return {JSX}         SelectInput Component
*/
const FloatingActionButton = ({ className, icon, onClick, position="-bottom-right" }) => {
	return (
		<div className={`floatingActionButton ${position} ${className ? className : ""}`} icon={icon} onClick={onClick}>
            <img src={icon}/>
		</div>
	);
};

FloatingActionButton.propTypes = {
    className: PropTypes.string,
    icon: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    position: PropTypes.string
};

export default FloatingActionButton;