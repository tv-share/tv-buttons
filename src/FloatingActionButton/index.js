import "./floatingActionButton.*";
import FloatingActionButton from "./FloatingActionButton";

export default FloatingActionButton;