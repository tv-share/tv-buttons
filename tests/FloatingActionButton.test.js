import React from 'react';
import renderer from 'react-test-renderer';
import FloatingActionButton from "../src/lib/FloatingActionButton";

const fn = () => { };
const icon = "test";

it('renders absolute', () => {
	const component = renderer.create(
		<FloatingActionButton onClick={fn} icon={icon}/>
	);

	expect(component.toJSON()).toMatchSnapshot();
});

it('renders fixed', () => {
    const component = renderer.create(
        <FloatingActionButton fixed onClick={fn} icon={icon} />
    );

    expect(component.toJSON()).toMatchSnapshot();
});