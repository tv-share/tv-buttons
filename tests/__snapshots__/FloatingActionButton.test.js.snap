// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders absolute 1`] = `
<div
  className="tv-floating-action-button  "
  icon="test"
  onClick={[Function]}
>
  <img
    src="test"
  />
</div>
`;

exports[`renders fixed 1`] = `
<div
  className="tv-floating-action-button -fixed "
  icon="test"
  onClick={[Function]}
>
  <img
    src="test"
  />
</div>
`;
